libcxx/test/libcxx/utilities/utility/utility.intcmp/cmp_equal.codegen.sh.cpp

```diff
@@ -0,0 +1,78 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// This test is checking the LLVM IR
+// REQUIRES: clang
+
+// RUN: %{cxx} %s %{compile_flags} -O3 -c -S -emit-llvm -o - | %{check-output}
+
+#include <cstdint>
+#include <utility>
+
+bool cmp_equal_i16_i16(int16_t lhs, int16_t rhs) {
```

Comment:
I don't know that these tests are very useful, and I'm sure they're very non-portable.

It seems like we're testing the compiler and optimizer, rather than the library.

Reply:
I would agree that `cmp_equal` doesn't seem like the kind of thing that we *must* have codegen checks for. I'm much happier with e.g. the `std::copy` ones though.